response parsing, object classes for entities, update documentation and tests

sandeepm96 opened the pull request:
So I modified the code to have only one `GraphResponse` class. Furthermore the function `_parse` in `ResponseObject` class can now parse the response of all endpoints (graphs, layouts and groups), be it a single or multiple response.

sandeepm96 commented:
Also added extensibility for parsing response of delete request. The response object of delete request will only have the attribute `message`.

adbharadwaj commented on graphspace_python/api/endpoint/graphs.py:
```diff
@@ -109,12 +122,14 @@ def delete_graph(self, name):
 		else:
-			graph_by_id_path = GRAPHS_PATH + str(graph['id'])
-			return self.client._make_request("DELETE", graph_by_id_path).json()
+			graph_by_id_path = GRAPHS_PATH + str(response.graph.id)
```
There is no need to create a variable for this. We can directly append id to paths while making requests.

sandeepm96 replied:
I think this would be more clean and understandable. Should we not do this way?

sandeepm96 commented:
I have updated the test code and added docstrings and comments for better understandability.

sandeepm96 commented:
Implemented layout response parsing similar to graph response parsing and included in this PR as they are dependent on the `ResponseObject` module. Also updated the corresponding test methods.

adbharadwaj commented on graphspace_python/api/endpoint/layouts.py:
```diff
@@ -98,7 +99,9 @@ def post_graph_layout(self, graph_id, layout_name, positions_json=None, style_js
+			self.client._make_request("POST", layouts_path, data=data).json()
+		)
 
 	def update_graph_layout(self, graph_id, layout_id, layout_name=None, positions_json=None, style_json=None, is_shared=None):
```
update and post api should accept layout object instead of so many arguments

sandeepm96 replied:
Ok. will update that.

adbharadwaj commented on graphspace_python/api/endpoint/layouts.py:
```diff
@@ -188,7 +191,9 @@ def update_graph_layout(self, graph_id, layout_id, layout_name=None, positions_j
 		layout_by_id_path = LAYOUTS_PATH.format(graph_id) + str(layout_id)
-		return self.client._make_request("PUT", layout_by_id_path, data=data).json()
+		return LayoutResponse(
+			self.client._make_request("PUT", layout_by_id_path, data=data).json()
```
Do not convert the response in json format...instead return the layout object.

sandeepm96 replied:
Here the layout object is being returned after parsing the json data from the response...anyways we have to get the json data from the response first to parse it

adbharadwaj commented on tests/test.py:
```diff
@@ -89,74 +89,74 @@ def test_post_graph(name=None):
 	graphspace.set_api_host('localhost:8000')
 	response = graphspace.get_my_graph_layouts(graph_id=graph_id)
-	assert response is not None and len(response['layouts']) >= 0
+	assert hasattr(response, 'layouts') and len(response.layouts) >= 0
```
response should be of type Layout class object (add a check for that too in the test code).

sandeepm96 replied:
Okay.

sandeepm96 replied:
Done and also divided the test file into separate files for each endpoint.

adbharadwaj commented:
Looks good...but can you add the tutorial for layouts as well?

sandeepm96 commented:
Yes. I am working on that now and will update it soon.

adbharadwaj commented on graphspace_python/api/endpoint/layouts.py:
```diff
@@ -1,225 +1,91 @@
 from graphspace_python.api.config import LAYOUTS_PATH
+from graphspace_python.api.obj.layout_response import LayoutResponse
 
 class Layouts(object):
```
Add pydoc.

sandeepm96 replied:
Done!

adbharadwaj commented on graphspace_python/api/endpoint/layouts.py:
```diff
@@ -1,225 +1,91 @@
 		self.client = client
 
-	def post_graph_layout(self, graph_id, layout_name, positions_json=None, style_json=None, is_shared=None):
+	def post_graph_layout(self, graph_id, layout, is_shared=None):
```
Make `is_shared` a member attribute of GSLayout class.

adbharadwaj replied:
Similarly, update the GSGraph class - include `is_public` attribute.

adbharadwaj commented on graphspace_python/api/endpoint/layouts.py:
```diff
@@ -1,225 +1,91 @@
-
+		:return: LayoutResponse object that wraps the response.
 		"""
 		data = {
```
Write a `__dict__` method for GSLayout class.
